users.js

```diff
@@ -616,7 +616,7 @@ class User {
 			targetUser = target;
 		}
 
-		if (room && room.auth) {
+		if (room && (room.auth || room.parent && room.parent.auth)) {
```

Comment:
There's no need to check `room.parent.auth`. What if a parent's parent has auth? `getAuth` will already do all this for you.

```suggestion
		if (room && (room.auth || room.parent)) {
```

Reply:
would a simple if (room) check be enough? Since whether or not the room has auth object is checked in ``room#getAuth`` anyways, and if it doesnt the global rank is returned regardless